mbed_targets/mbed_targets.py

```diff
@@ -181,15 +164,13 @@ def json_dump(self) -> str:
         return json.dumps([asdict(t) for t in self], indent=4)
 
 
-def _get_target(query: TargetDatabaseQuery, mode: DatabaseMode = DatabaseMode.AUTO) -> MbedTarget:
-    if mode == DatabaseMode.OFFLINE:
+def _get_target(query: TargetDatabaseQuery) -> MbedTarget:
+    if MBED_DATABASE_MODE == DatabaseMode.OFFLINE:
         return MbedTargets.from_offline_database().get_target(**query)
-    if mode == DatabaseMode.ONLINE:
+    elif MBED_DATABASE_MODE == DatabaseMode.ONLINE:
```

Comment:
Why do we need `elif` here? Another thing I noticed which isn't part of this PR, we have a redundant `else` clause lower down.

Reply:
Am I correct in understanding that you're suggesting:
```
if a:
  return ..
if b:
  return ...
return ...
```
?

Reply:
I am. We return in each branch so we don't need `elif` or `else`

Reply:
https://github.com/ARMmbed/mbed-targets/pull/102/commits/5f6f6f9c3a1285ded2b8992d47b38220f171ba6c